### DISCUSSION TITLE
JHipster-JDL created entity/table PK autoincrement NO; how to change to YES

### DISCUSSION THREAD

**Discussion by djsg** [jhipster/generator-jhipster#27865] [Q&A]
I used jhipster 8.7.1 to created a Spring Boot app. My database is PostgreSQL.

I found that autoincrement of primary key of jdl-created entity tables is NO.

 - I wonder how does the Spring framework handle the primary key when doing table insertion.

I did a insertion test, the table ID had a big jump from 10 to 1000+, which seems not very good.

 - Is there a way to make primary key autoincrement YES for jdl-created table?

**Comment by TatyOko28** [comment]
What you are seeing is expected behavior in JHipster + PostgreSQL.

JHipster does **not** usually use the old PostgreSQL `SERIAL` auto-increment style.  
Instead, it uses **database sequences** together with JPA/Hibernate.

So even if pgAdmin shows:

```text
Autoincrement = NO
